CompilerSource/parsing/ast.h

```diff
@@ -508,80 +524,92 @@ class AST {
-    virtual bool VisitInitializer(Initializer &node);
-    virtual bool VisitNewExpression(NewExpression &node);
-    virtual bool VisitDeleteExpression(DeleteExpression &node);
-    virtual bool VisitDeclarationStatement(DeclarationStatement &node);
+    virtual bool VisitCodeBlock(CodeBlock &node){ return DefaultVisit(node); }
+    virtual bool VisitBinaryExpression(BinaryExpression &node){ return DefaultVisit(node); }
+    virtual bool VisitFunctionCallExpression(FunctionCallExpression &node){ return DefaultVisit(node); }
+    virtual bool VisitUnaryPrefixExpression(UnaryPrefixExpression &node){ return DefaultVisit(node); }
+    virtual bool VisitUnaryPostfixExpression(UnaryPostfixExpression &node){ return DefaultVisit(node); }
+    virtual bool VisitTernaryExpression(TernaryExpression &node){ return DefaultVisit(node); }
+    virtual bool VisitLambdaExpression(LambdaExpression &node){ return DefaultVisit(node); }
+    virtual bool VisitSizeofExpression(SizeofExpression &node){ return DefaultVisit(node); }
+    virtual bool VisitAlignofExpression(AlignofExpression &node){ return DefaultVisit(node); }
+    virtual bool VisitCastExpression(CastExpression &node){ return DefaultVisit(node); }
+    virtual bool VisitParenthetical(Parenthetical &node){ return DefaultVisit(node); }
+    virtual bool VisitArray(Array &node){ return DefaultVisit(node); }
+    virtual bool VisitIdentifierAccess(IdentifierAccess &node){ return DefaultVisit(node); }
+    virtual bool VisitLiteral(Literal &node){ return DefaultVisit(node); }
+    virtual bool VisitIfStatement(IfStatement &node){ return DefaultVisit(node); }
+    virtual bool VisitForLoop(ForLoop &node){ return DefaultVisit(node); }
+    virtual bool VisitWhileLoop(WhileLoop &node){ return DefaultVisit(node); }
+    virtual bool VisitDoLoop(DoLoop &node){ return DefaultVisit(node); }
+    virtual bool VisitCaseStatement(CaseStatement &node){ return DefaultVisit(node); }
+    virtual bool VisitDefaultStatement(DefaultStatement &node){ return DefaultVisit(node); }
+    virtual bool VisitSwitchStatement(SwitchStatement &node){ return DefaultVisit(node); }
+    virtual bool VisitReturnStatement(ReturnStatement &node){ return DefaultVisit(node); }
+    virtual bool VisitBreakStatement(BreakStatement &node){ return DefaultVisit(node); }
+    virtual bool VisitContinueStatement(ContinueStatement &node){ return DefaultVisit(node); }
+    virtual bool VisitWithStatement(WithStatement &node){ return DefaultVisit(node); }
+    virtual bool VisitInitializer(Initializer &node){ return DefaultVisit(node); }
+    virtual bool VisitNewExpression(NewExpression &node){ return DefaultVisit(node); }
+    virtual bool VisitDeleteExpression(DeleteExpression &node){ return DefaultVisit(node); }
+    virtual bool VisitDeclarationStatement(DeclarationStatement &node){ return DefaultVisit(node); }
     virtual bool Visit(PNode &node) {
       return node->accept(*this);
     }
+  };
 
-    virtual ~Visitor() { of.close(); }
```

Comment:
If I did that, some files used during the compilation process would be closed and not reopened, which would lead to errors.